### ISSUE TITLE
GitHub checks CI pipeline fails on Windows build

### ISSUE THREAD

**Issue by crtag** [atomCAD/atomCAD#41]
This prevents other PR validation checks CIs execution as well.

**The following warnings were emitted during compilation:**

> **warning**: shaderc: searching for native shaderc libraries on system;  
> use `--features build-from-source` to force building from source code

> **warning**: shaderc: cannot find native shaderc library on system;  
> falling back to build from source

**error**: failed to run custom build command for `shaderc-sys v0.8.2`

**Caused by**:  
process didn't exit successfully: `D:\a\atomCAD\atomCAD\target\debug\build\shaderc-sys-8502e2f50acd6682\build-script-build` (exit code: 101)

**--- stdout**  
> **cargo:warning**=shaderc: searching for native shaderc libraries on system;  
> use `--features build-from-source` to force building from source code

> **cargo:warning**=shaderc: cannot find native shaderc library on system;  
> falling back to build from source

**--- stderr**  
thread 'main' panicked at `C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\shaderc-sys-0.8.2\build\cmd_finder.rs:55:13`:  

> couldn't find required command: "ninja"

> **note**: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**warning**: build failed, waiting for other jobs to finish...  
**Error**: Process completed with exit code 1.

**Comment by maaku** [comment]
Need to add the ninja build tool to the CI workflow: https://github.com/marketplace/actions/setup-ninja

no idea how this was working previously tbh

**Comment by crtag** [comment]
Adding the following step fixes the issue. Windows test went fine, see results in https://github.com/atomCAD/atomCAD/pull/40 and please ignore other linter errors

```
      - name: Setup Ninja
        if: runner.os == 'windows'
        uses: ashutoshvarma/setup-ninja@master
        with:
          version: 1.11.1
          dest: ${{ github.workspace }}/ninja_bin
```
@maaku Would you prefer a separate PR addressing this issue so it's tracked granularly?

**Comment by maaku** [comment]
Fixed in #42
